src/d3a/models/strategy/area_agents/one_sided_engine.py

```diff
@@ -143,7 +143,13 @@ def event_trade(self, *, trade):
                 f"offer: source_rate ({source_rate}) is not lower than target_rate ({target_rate})"
 
             try:
-                trade_offer_rate = trade.offer.energy_rate
+                if ConstSettings.IAASettings.MARKET_TYPE == 1 and trade is not None:
```

Comment:
I think the second check `..and trade is not None` could be avoided as at its been validated at R138 at this point of commit ?

Reply:
Good catch. Fixing it, thanks!